### PR TITLE
Simple refactoring on tree-view.js decaf

### PR DESCRIPTION
**Pull request by mauricioszabo**
I had already decaffed this file in the `feature/latest-electron` version with the Decaffeinate project. I feel that it generated a better JS than the one that's present on Master :)

### CODE REVIEW

**Comment by confused-Techie** [comment]
@mauricioszabo I know the JS in `tree-view.js` decaf is messy, it's because I couldn't get the tests to pass unless I decaffed specifically CoffeeScript `1.12.7`. Which unfortunately we can see tests are not passing here.

But this is exactly what I wanted to do after that initial PR. Make a PR just trying to cleanup this file to determine what's going wrong with the decaf work on this on.

So if it's fine by you, I'll grab this new branch and see if I can't see what's causing those tests to fail, so that we can have functional and nice to look at JavaScript.

Thanks for making this one

**Comment by schadomi7** [comment]
Hi, I am trying to understand tree-view for teletype integration https://github.com/pulsar-edit/pulsar/issues/536#issuecomment-1979230168 and I am having a terrible time with it. However, this PR is much more readable. 
I'll like to try and help. Unfortunately I cannot see the logs, as they expired. Can someone rerun the tests?

**Comment by confused-Techie** [comment]
@schadomi7 I took a look to rerun the logs here, but [apparently](https://docs.github.com/en/actions/managing-workflow-runs/re-running-workflows-and-jobs) GitHub Actions can only be rerun for as long as the duration of their log retention. A strange choice for sure, but that means it may be better off running these logs locally.

Although, as the one that committed this barely decaffed code in Tree-View originally, I know it's not great, but there's something about the decaf process that seems to completely break this file, and I was never able to find exactly what it was. That could be something I'll take a better look at now on not as much of a time crunch.

But if you are just wanting to have a look at a less confusing, functional `tree-view` src file, [here](https://github.com/pulsar-edit/pulsar/commit/348d23095cb12af615ccc2536fd6349272592b03)s the same file right before I decaffed it. It should be identical to the file we have in `master` except it's written in CoffeeScript, prior it being decaffed

**Comment by schadomi7** [comment]
@confused-Techie Thank you! Did not know that about Github Actions.

Sounds great, I do not yet know exactly what needs to happen for teletype to be able to sync the tree-view, but if it will require changes in the tree-view, a more readable tree-view is appreciated.
I'll take a look at the Coffeescript file in the meantime.

I've been having some trouble getting the tests to run locally, and I was too lazy to debug it properly, so far. Welp, no use in delaying further..

File: packages/tree-view/lib/tree-view.js [hunk 1]
```diff
@@ -1,2048 +1,1583 @@
...
+    this.disposables.add(this.onDeleteEntryFailed(({pathToDelete}) => {
+      const index = this.editorsToDestroy.indexOf(pathToDelete);
+      if (index !== -1) { return this.editorsToDestroy.splice(index, 1); }
+    })
+    );
+  }
+
+  serialize() {
+    return {
+      directoryExpansionStates: new (function(roots) {
```
Comment (Psychosynthesis) [inline packages/tree-view/lib/tree-view.js:202]: It’s not that I’m being picky... But I don’t understand why it’s done so strangely here? What prevented you from making this place through a simple search?

I mean that in this place a new instance of the function will be created each time, this does not look very correct.

Perhaps I do not understand some nuance, I apologize in advance.

File: packages/tree-view/lib/tree-view.js [hunk 2]
```diff
@@ -1,2048 +1,1583 @@
...
-    TreeView.prototype.handleEvents = function() {
-      this.element.addEventListener('click', (function(_this) {
-        return function(e) {
-          // This prevents accidental collapsing when a .entries element is the event target
-          if (e.target.classList.contains('entries')) {
-            return;
+      this.roots = (() => {
+        const result = [];
+        for (let projectPath of Array.from(projectPaths)) {
+          var left;
```
Comment (Psychosynthesis) [inline packages/tree-view/lib/tree-view.js:469]: Why here is a `var` not `let`?

As far as I can see, according to the file, this variable is used only within this block; another one already defines its own.

File: packages/tree-view/lib/tree-view.js [hunk 3]
```diff
@@ -1,2048 +1,1583 @@
...
-            return function() {
-              return _this.currentlyOpening["delete"](filePath);
-            };
-          })(this));
-        }
-      } else if (detail === 2) {
-        return this.openAfterPromise(filePath, {
-          searchAllPanes: alwaysOpenExisting
-        });
+      if (nextEntry = this.nextEntry(selectedEntry)) {
```
Comment (Psychosynthesis) [inline packages/tree-view/lib/tree-view.js:595]: `==` here?

File: packages/tree-view/lib/tree-view.js (hunk 1 above)
Comment (mauricioszabo) [reply packages/tree-view/lib/tree-view.js:202]: This I believe was auto-generated. It was quite a long time ago...

Comment (Psychosynthesis) [reply packages/tree-view/lib/tree-view.js:202]: But maybe it’s still worth fixing this a little at the same time, since this is a refactoring?

**Comment by confused-Techie** [comment]
With #1052 merged it seems we are good to go ahead and close this one, is that alright @mauricioszabo?

**Comment by mauricioszabo** [comment]
Sure, sounds good to me :)

**Pull request closed without merging**